sklearn/covariance/_shrunk_covariance.py

```diff
@@ -686,14 +691,19 @@ def fit(self, X, y=None):
         self._validate_params()
 
         X = self._validate_data(X)
+        if X.ndim == 1:
+            X = np.reshape(X, (1, -1))
+            warnings.warn(
```

Comment:
While writing unit tests for `X.ndim=1` I found 1d arrays have been deprecated and `X = self._validate_data(X)` throws the following error.
![Screenshot from 2022-11-16 13-54-50](https://user-images.githubusercontent.com/30499524/202212347-516182a7-36e9-487c-9e5f-92af027ed997.png)

**Solution:**

1.  Either we can move the check `if X.ndim == 1:` above `X = self._validate_data(X)` and add unit test for `X.ndim=1`. In this two warnings will be raised.

![Screenshot from 2022-11-16 14-52-43](https://user-images.githubusercontent.com/30499524/202213762-9b7d8ce5-2798-4206-a234-4094a0416882.png)

(OR)
3. May be we can remove this check.
(OR)
4. We can ignore the coverage for this statement and leave it unchanged. I sort followed this [PR](https://github.com/scikit-learn/scikit-learn/pull/24870) implementation.